MyBotLogic/MissionManager.cpp

```diff
@@ -0,0 +1,27 @@
+#include "MissionManager.h"
+
+MissionManager MissionManager::m_instance;
+
+
+void MissionManager::subscribeGoal(unsigned fromZone, unsigned destinationZone)
+{
+    m_objectives.push_back(std::make_pair(fromZone, destinationZone));
+    //Consulter Planificateur pour obtenir les sous missions
+    retourPlanificateur stepsToReachGoal = Planificateur::createMissionsToReachGoal(fromZone, destinationZone);
+    m_missions.insert(m_missions.end(), stepsToReachGoal.begin(), stepsToReachGoal.end());
+}
+
+void MissionManager::subscribeMission(Mission mission, unsigned int npcId)
+{
+    
+    mission.setId(m_missionsCount);
+    ++m_missionsCount;
+    //Consulter Planificateur pour obtenir les sous missions
+    retourPlanificateur subMissions = Planificateur::planifiedMissions(mission, npcId);
+    for (Mission submis : subMissions)
+    {
+        mission.setId(m_missionsCount);
```

Comment:
Actuellement cette opération s'applique uniquement sur la variable mission et non sur la variable submis de ta boucle for, donc on fait une boucle pour rien dans la cas présent :( Juste remplacer `mission` par `submis` corrigera le soucis.
Ta remarque est bonne mais je ne pense pas que le planificateur ne devrait pas retourner la mission mère (qui devrait être un objectif plutôt non ?)